### PR TITLE
Config files, formatting, error linting, Auth refactoring, Folder structure etc.

### PR DESCRIPTION
**Pull request by bsienn-khan**
### Commit 1
d5adfad6231ae16ae60d2ab9d72712c68611dade

Files added as follows:

> The following 4 things help code consistency and writhing clean & less buggy code.

**.editorconfig** (RP1)
This helps with syncing editor configs so that everyone has the same settings to avoid inconsistency
I've put very minimum settings just 3 main ones
Reference: https://editorconfig.org

**.eslintrc**  (RP2)
This helps reduce errors in code, i.e. when a loop is used and a key attribute is missing or is invalid.
There are a few issues with our code, so this would help fix the,
I've added a script/command to auto-fix fixable issues, run `yarn lint`
By the way, after installing required plugins, the listing can be automated on each save of a file. 
**Very handy and highly insisted to enable in the editor settings**.
Note: As we have dedicated the linting file, thus I've moved some config from package.json to this file.

**.prettierignore && .prettier.json** (RP3)
Highly recommended to add `prettier` to format code, It is opinionated by the author of the plugin but this is widely used in many projects around the js ecosystem.
I've added some overrides to the default set by the plugin, the rest of the settings are handled by the plugin itself.

**jsconfig.json**
It's a bit more technical to describe in a comment, so please see the provided link.
Reference: https://code.visualstudio.com/docs/languages/jsconfig

*Required Plugins (RP)
https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig
https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

### CODE REVIEW

**Comment by ARehmanMahi** [comment]
### Commit 2
After running `yarn lint` that auto fixes the fixable issues, I manually fixed the remaining errors,
Pic as a reference of `yarn lint` result

Screenshot from 2022-09-22 16-15-31

**Comment by ARehmanMahi** [comment]
### Commit 3

Packages added:

Material Fonts
@fontsource/roboto

Material Icons
@mui/icons-material

https://reactjs.org/docs/faq-styling.html#what-is-css-in-js
classnames

https://mui.com/material-ui/discover-more/related-projects/#image
mui-image

https://immerjs.github.io/immer/example-setstate#useimmer
use-immer

**Comment by ARehmanMahi** [comment]
### Commit 5
When we use Lazy loading of the components/routes, the component gets loaded asynchronously.
And since async call of the route doesn't show up until the file is loaded, we would get "blank page until it does"
What I actually mean by that is that we need to handle the async loading, and have a way to show something else like a loader instead of a blank page.

That's where the `Suspense` part comes in and, by the way, It is necessary to use `Suspense` for component lazy routes, it's not optional.

**Comment by ARehmanMahi** [comment]
### Commit 7

@Ian-arap-kirui 
I moved the `RequireAuth` logic from App.jsx in its own component but, made use of context-api instead to dynamically (on each page load) get the status of the user.

just use `const { user } = useAuth();` and if the object is not null then it mean the user is logged in. You can get user info from that user object.

Make sure to import `import { useAuth } from './contexts/AuthContext';` before calling that hook in component.
This would be available in the whole app, awkwardly also on the customer side, but frankly, I don't know yet how to make it work only for the admin part. Pull requests are welcome for that.

**Comment by ARehmanMahi** [comment]
### Commit 8

There is a caveat using lazy loading on routes explained in the docs.
TLDR:

instead of using the routers `Navigate` hook for navigation, use an alternative hook, which you can find in the hooks folder ready to use. It's nothing fency, it just use the pattern react provides when dealing with lazy loading routes/components.

Example:
```js

import { useTransNavigate } from "../../hooks/useTransNavigate";

const MainMenu = () => {
  // const Navigate = useNavigate(); <= old code
 const [isPending, transNavigate] = useTransNavigate();

<MenuItem onClick={ () => { transNavigate("/account"); // Navigate("/account") <= old code }}>
```
